extensions/pyRevitTools.extension/checks/radar_check.py

```diff
@@ -0,0 +1,624 @@
+                                            value=distance,
+                                            forEditing=True)
+    else:
+        ui_units = DB.UnitFormatUtils.Format(units=document.GetUnits(),
+                                             unitType=DB.UnitType.UT_Length, 
+                                             value=distance, maxAccuracy=False, 
+                                             forEditing=True)
+    return ui_units
+
+def calculate_distance(point1, point2):
+    """
+    Calculates the distance between two points.
+    Args:
+        point1 (tuple): The coordinates of the first point.
+        point2 (tuple): The coordinates of the second point.
+    Returns:
+        distance (float): The distance between the two points."""
+    # Unpack the tuples
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+    # Calculate the distance using the Euclidean distance formula
+    distance =( #rounded to the nearest inch
+        math.sqrt((x2 - x1)**2 + (y2 - y1)**2 + (z2 - z1)**2))
+    return distance
+
+def calculate_horizontal_distance(point1, point2):
+    """Calculates the horizontal distance between two points."""
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+    distance = math.sqrt((x2 - x1)**2 + (y2 - y1)**2)
+    return distance
+
+def calculate_delta_x(point1, point2):
+    """Calculates the delta X between two points."""
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+    delta_x = x2 - x1
+    return delta_x
+
+def calculate_delta_y(point1, point2):
+    """Calculates the delta Y between two points."""
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+    delta_y = y2 - y1
+    return delta_y
+
+def calculate_angle(point1, point2):
+    """
+    Calculates the angle between two points.
+    Args:
+        point1 (tuple): The coordinates of the first point.
+        point2 (tuple): The coordinates of the second point.
+    Returns:
+        angle (float): The angle between the two points."""
+    # Unpack the tuples
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+    # Calculate the angle using the arctangent function
+    angle = round(math.degrees(math.atan2(y2 - y1, x2 - x1)), 2)
+    return angle
+
+def check_bounding_box(bbox, intorig, extentdistance):
+    """
+    Checks if the domain of the bounding box is more than 10
+     miles from the internal origin.
+     Args:
+        bbox (BoundingBoxXYZ): The bounding box to check.
+        intorig (tuple): The coordinates of the internal origin.
+        extentdistance (float): The distance from the internal origin.
+    Returns:
+        Status (int): The status of the bounding box."""
+    min = (bbox.Min.X, bbox.Min.Y, bbox.Min.Z)
+    max = (bbox.Max.X, bbox.Max.Y, bbox.Max.Z)
+    if (calculate_distance(min, intorig) > extentdistance or 
+        calculate_distance(max, intorig) > extentdistance):
+        Status = 0
+    else:
+        Status = 1
+    return Status
+
+def get_project_base_and_survey_pts(document=doc):
```

Comment:
split into 2 functions, one for the base point and one for the survey point.
Return the point (XYZ) directly (see my other comments)